util/subtree_darwin.go

```diff
@@ -46,17 +46,23 @@ func cleanup(key string, logger grip.Journaler) error {
 		}
 		splitLine := strings.Fields(line)
 		pid := splitLine[0]
+		command := splitLine[1]
 		env := splitLine[2:]
 
-		if pid != myPid && envHasMarkers(key, env) {
-			// add it to the list of processes to clean up
-			pidAsInt, err := strconv.Atoi(pid)
-			if err != nil {
-				logger.Errorf("cleanup failed to convert from string to int: %v", err)
-				continue
-			}
-			pidsToKill = append(pidsToKill, pidAsInt)
+		if pid == myPid {
+			continue
+		}
+		if !envHasMarkers(key, env) && !strings.HasPrefix(command, workingDir) {
```

Comment:
Is there a risk that `workingDir` will be somehow unset and the agent kills way more than it intends to? I wonder if there's some sanity check we can add somewhere to reduce the risk of that. Maybe just check that `workingDir` is not the empty string?

Reply:
done